<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_24) on Sun Apr 08 20:51:54 UTC 2012 -->
<META http-equiv="Content-Type" content="text/html; charset=UTF-8">
<TITLE>
NEATGenome (Encog Core 3.1.0 API)
</TITLE>

<META NAME="date" CONTENT="2012-04-08">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="NEATGenome (Encog Core 3.1.0 API)";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/NEATGenome.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV CLASS&nbsp;
&nbsp;<A HREF="../../../../../org/encog/neural/neat/training/NEATInnovation.html" title="class in org.encog.neural.neat.training"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../index.html?org/encog/neural/neat/training/NEATGenome.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="NEATGenome.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
org.encog.neural.neat.training</FONT>
<BR>
Class NEATGenome</H2>
<PRE>
<A HREF="http://java.sun.com/javase/6/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</A>
  <IMG SRC="../../../../../resources/inherit.gif" ALT="extended by "><A HREF="../../../../../org/encog/ml/genetic/genome/BasicGenome.html" title="class in org.encog.ml.genetic.genome">org.encog.ml.genetic.genome.BasicGenome</A>
      <IMG SRC="../../../../../resources/inherit.gif" ALT="extended by "><B>org.encog.neural.neat.training.NEATGenome</B>
</PRE>
<DL>
<DT><B>All Implemented Interfaces:</B> <DD><A HREF="http://java.sun.com/javase/6/docs/api/java/io/Serializable.html?is-external=true" title="class or interface in java.io">Serializable</A>, <A HREF="http://java.sun.com/javase/6/docs/api/java/lang/Cloneable.html?is-external=true" title="class or interface in java.lang">Cloneable</A>, <A HREF="http://java.sun.com/javase/6/docs/api/java/lang/Comparable.html?is-external=true" title="class or interface in java.lang">Comparable</A>&lt;<A HREF="../../../../../org/encog/ml/genetic/genome/Genome.html" title="interface in org.encog.ml.genetic.genome">Genome</A>&gt;, <A HREF="../../../../../org/encog/ml/genetic/genome/Genome.html" title="interface in org.encog.ml.genetic.genome">Genome</A></DD>
</DL>
<HR>
<DL>
<DT><PRE>public class <B>NEATGenome</B><DT>extends <A HREF="../../../../../org/encog/ml/genetic/genome/BasicGenome.html" title="class in org.encog.ml.genetic.genome">BasicGenome</A><DT>implements <A HREF="http://java.sun.com/javase/6/docs/api/java/lang/Cloneable.html?is-external=true" title="class or interface in java.lang">Cloneable</A>, <A HREF="http://java.sun.com/javase/6/docs/api/java/io/Serializable.html?is-external=true" title="class or interface in java.io">Serializable</A></DL>
</PRE>

<P>
Implements a NEAT genome. This is a "blueprint" for creating a neural
 network.

 NeuroEvolution of Augmenting Topologies (NEAT) is a genetic algorithm for the
 generation of evolving artificial neural networks. It was developed by Ken
 Stanley while at The University of Texas at Austin.

 http://www.cs.ucf.edu/~kstanley/
<P>

<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../../../serialized-form.html#org.encog.neural.neat.training.NEATGenome">Serialized Form</A></DL>
<HR>

<P>
<!-- =========== FIELD SUMMARY =========== -->

<A NAME="field_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Field Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="http://java.sun.com/javase/6/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#PROPERTY_LINKS">PROPERTY_LINKS</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;<A HREF="http://java.sun.com/javase/6/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#PROPERTY_NEURONS">PROPERTY_NEURONS</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#TWEAK_DISJOINT">TWEAK_DISJOINT</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The adjustment factor for disjoint genes.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#TWEAK_EXCESS">TWEAK_EXCESS</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The adjustment factor for excess genes.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#TWEAK_MATCHED">TWEAK_MATCHED</A></B></CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The adjustment factor for matched genes.</TD>
</TR>
</TABLE>
&nbsp;
<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#NEATGenome()">NEATGenome</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#NEATGenome(long, org.encog.ml.genetic.genome.Chromosome, org.encog.ml.genetic.genome.Chromosome, int, int)">NEATGenome</A></B>(long&nbsp;genomeID,
           <A HREF="../../../../../org/encog/ml/genetic/genome/Chromosome.html" title="class in org.encog.ml.genetic.genome">Chromosome</A>&nbsp;neurons,
           <A HREF="../../../../../org/encog/ml/genetic/genome/Chromosome.html" title="class in org.encog.ml.genetic.genome">Chromosome</A>&nbsp;links,
           int&nbsp;inputCount,
           int&nbsp;outputCount)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Create a NEAT gnome.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#NEATGenome(long, int, int)">NEATGenome</A></B>(long&nbsp;id,
           int&nbsp;inputCount,
           int&nbsp;outputCount)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Construct a genome, do not provide links and neurons.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#NEATGenome(org.encog.neural.neat.training.NEATGenome)">NEATGenome</A></B>(<A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html" title="class in org.encog.neural.neat.training">NEATGenome</A>&nbsp;other)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Construct a genome by copying another.</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#alreadyHaveThisNeuronID(long)">alreadyHaveThisNeuronID</A></B>(long&nbsp;id)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Do we already have this neuron id?</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#decode()">decode</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Convert the genes to an actual network.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#encode()">encode</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Convert the network to genes.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#getCompatibilityScore(org.encog.neural.neat.training.NEATGenome)">getCompatibilityScore</A></B>(<A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html" title="class in org.encog.neural.neat.training">NEATGenome</A>&nbsp;genome)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the compatibility score with another genome.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#getInputCount()">getInputCount</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/encog/ml/genetic/genome/Chromosome.html" title="class in org.encog.ml.genetic.genome">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#getLinks()">getLinks</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/encog/ml/genetic/genome/Chromosome.html" title="class in org.encog.ml.genetic.genome">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#getLinksChromosome()">getLinksChromosome</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#getNetworkDepth()">getNetworkDepth</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/encog/ml/genetic/genome/Chromosome.html" title="class in org.encog.ml.genetic.genome">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#getNeurons()">getNeurons</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../org/encog/ml/genetic/genome/Chromosome.html" title="class in org.encog.ml.genetic.genome">Chromosome</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#getNeuronsChromosome()">getNeuronsChromosome</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#getNumGenes()">getNumGenes</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#getOutputCount()">getOutputCount</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;long</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#getSpeciesID()">getSpeciesID</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;double</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#getSplitY(int)">getSplitY</A></B>(int&nbsp;nd)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get the specified split y.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;boolean</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#isDuplicateLink(long, long)">isDuplicateLink</A></B>(long&nbsp;fromNeuronID,
                long&nbsp;toNeuronID)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Determine if this is a duplicate link.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#mutateActivationResponse(double, double)">mutateActivationResponse</A></B>(double&nbsp;mutateRate,
                         double&nbsp;maxPertubation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Mutate the activation response.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#mutateWeights(double, double, double)">mutateWeights</A></B>(double&nbsp;mutateRate,
              double&nbsp;probNewMutate,
              double&nbsp;maxPertubation)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Mutate the weights.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#setInputCount(int)">setInputCount</A></B>(int&nbsp;inputCount)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#setLinksChromosome(org.encog.ml.genetic.genome.Chromosome)">setLinksChromosome</A></B>(<A HREF="../../../../../org/encog/ml/genetic/genome/Chromosome.html" title="class in org.encog.ml.genetic.genome">Chromosome</A>&nbsp;linksChromosome)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#setNetworkDepth(int)">setNetworkDepth</A></B>(int&nbsp;networkDepth)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#setNeuronsChromosome(org.encog.ml.genetic.genome.Chromosome)">setNeuronsChromosome</A></B>(<A HREF="../../../../../org/encog/ml/genetic/genome/Chromosome.html" title="class in org.encog.ml.genetic.genome">Chromosome</A>&nbsp;neuronsChromosome)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#setOutputCount(int)">setOutputCount</A></B>(int&nbsp;outputCount)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#setSpeciesID(long)">setSpeciesID</A></B>(long&nbsp;species)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Set the species id.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#sortGenes()">sortGenes</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Sort the genes.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html#validate()">validate</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_org.encog.ml.genetic.genome.BasicGenome"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class org.encog.ml.genetic.genome.<A HREF="../../../../../org/encog/ml/genetic/genome/BasicGenome.html" title="class in org.encog.ml.genetic.genome">BasicGenome</A></B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><A HREF="../../../../../org/encog/ml/genetic/genome/BasicGenome.html#calculateGeneCount()">calculateGeneCount</A>, <A HREF="../../../../../org/encog/ml/genetic/genome/BasicGenome.html#compareTo(org.encog.ml.genetic.genome.Genome)">compareTo</A>, <A HREF="../../../../../org/encog/ml/genetic/genome/BasicGenome.html#equals(org.encog.ml.genetic.genome.Genome)">equals</A>, <A HREF="../../../../../org/encog/ml/genetic/genome/BasicGenome.html#getAdjustedScore()">getAdjustedScore</A>, <A HREF="../../../../../org/encog/ml/genetic/genome/BasicGenome.html#getAmountToSpawn()">getAmountToSpawn</A>, <A HREF="../../../../../org/encog/ml/genetic/genome/BasicGenome.html#getChromosomes()">getChromosomes</A>, <A HREF="../../../../../org/encog/ml/genetic/genome/BasicGenome.html#getGeneticAlgorithm()">getGeneticAlgorithm</A>, <A HREF="../../../../../org/encog/ml/genetic/genome/BasicGenome.html#getGenomeID()">getGenomeID</A>, <A HREF="../../../../../org/encog/ml/genetic/genome/BasicGenome.html#getOrganism()">getOrganism</A>, <A HREF="../../../../../org/encog/ml/genetic/genome/BasicGenome.html#getPopulation()">getPopulation</A>, <A HREF="../../../../../org/encog/ml/genetic/genome/BasicGenome.html#getScore()">getScore</A>, <A HREF="../../../../../org/encog/ml/genetic/genome/BasicGenome.html#mate(org.encog.ml.genetic.genome.Genome, org.encog.ml.genetic.genome.Genome, org.encog.ml.genetic.genome.Genome)">mate</A>, <A HREF="../../../../../org/encog/ml/genetic/genome/BasicGenome.html#setAdjustedScore(double)">setAdjustedScore</A>, <A HREF="../../../../../org/encog/ml/genetic/genome/BasicGenome.html#setAmountToSpawn(double)">setAmountToSpawn</A>, <A HREF="../../../../../org/encog/ml/genetic/genome/BasicGenome.html#setGeneticAlgorithm(org.encog.ml.genetic.GeneticAlgorithm)">setGeneticAlgorithm</A>, <A HREF="../../../../../org/encog/ml/genetic/genome/BasicGenome.html#setGenomeID(long)">setGenomeID</A>, <A HREF="../../../../../org/encog/ml/genetic/genome/BasicGenome.html#setOrganism(java.lang.Object)">setOrganism</A>, <A HREF="../../../../../org/encog/ml/genetic/genome/BasicGenome.html#setPopulation(org.encog.ml.genetic.population.Population)">setPopulation</A>, <A HREF="../../../../../org/encog/ml/genetic/genome/BasicGenome.html#setScore(double)">setScore</A>, <A HREF="../../../../../org/encog/ml/genetic/genome/BasicGenome.html#toString()">toString</A></CODE></TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.<A HREF="http://java.sun.com/javase/6/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</A></B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><A HREF="http://java.sun.com/javase/6/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</A>, <A HREF="http://java.sun.com/javase/6/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</A>, <A HREF="http://java.sun.com/javase/6/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</A>, <A HREF="http://java.sun.com/javase/6/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</A>, <A HREF="http://java.sun.com/javase/6/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</A>, <A HREF="http://java.sun.com/javase/6/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</A>, <A HREF="http://java.sun.com/javase/6/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</A>, <A HREF="http://java.sun.com/javase/6/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</A>, <A HREF="http://java.sun.com/javase/6/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</A>, <A HREF="http://java.sun.com/javase/6/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</A></CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ============ FIELD DETAIL =========== -->

<A NAME="field_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Field Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="PROPERTY_NEURONS"><!-- --></A><H3>
PROPERTY_NEURONS</H3>
<PRE>
public static final <A HREF="http://java.sun.com/javase/6/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</A> <B>PROPERTY_NEURONS</B></PRE>
<DL>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../../../constant-values.html#org.encog.neural.neat.training.NEATGenome.PROPERTY_NEURONS">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="PROPERTY_LINKS"><!-- --></A><H3>
PROPERTY_LINKS</H3>
<PRE>
public static final <A HREF="http://java.sun.com/javase/6/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</A> <B>PROPERTY_LINKS</B></PRE>
<DL>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../../../constant-values.html#org.encog.neural.neat.training.NEATGenome.PROPERTY_LINKS">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="TWEAK_DISJOINT"><!-- --></A><H3>
TWEAK_DISJOINT</H3>
<PRE>
public static final double <B>TWEAK_DISJOINT</B></PRE>
<DL>
<DD>The adjustment factor for disjoint genes.
<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../../../constant-values.html#org.encog.neural.neat.training.NEATGenome.TWEAK_DISJOINT">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="TWEAK_EXCESS"><!-- --></A><H3>
TWEAK_EXCESS</H3>
<PRE>
public static final double <B>TWEAK_EXCESS</B></PRE>
<DL>
<DD>The adjustment factor for excess genes.
<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../../../constant-values.html#org.encog.neural.neat.training.NEATGenome.TWEAK_EXCESS">Constant Field Values</A></DL>
</DL>
<HR>

<A NAME="TWEAK_MATCHED"><!-- --></A><H3>
TWEAK_MATCHED</H3>
<PRE>
public static final double <B>TWEAK_MATCHED</B></PRE>
<DL>
<DD>The adjustment factor for matched genes.
<P>
<DL>
<DT><B>See Also:</B><DD><A HREF="../../../../../constant-values.html#org.encog.neural.neat.training.NEATGenome.TWEAK_MATCHED">Constant Field Values</A></DL>
</DL>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="NEATGenome(org.encog.neural.neat.training.NEATGenome)"><!-- --></A><H3>
NEATGenome</H3>
<PRE>
public <B>NEATGenome</B>(<A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html" title="class in org.encog.neural.neat.training">NEATGenome</A>&nbsp;other)</PRE>
<DL>
<DD>Construct a genome by copying another.
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>other</CODE> - The other genome.</DL>
</DL>
<HR>

<A NAME="NEATGenome(long, org.encog.ml.genetic.genome.Chromosome, org.encog.ml.genetic.genome.Chromosome, int, int)"><!-- --></A><H3>
NEATGenome</H3>
<PRE>
public <B>NEATGenome</B>(long&nbsp;genomeID,
                  <A HREF="../../../../../org/encog/ml/genetic/genome/Chromosome.html" title="class in org.encog.ml.genetic.genome">Chromosome</A>&nbsp;neurons,
                  <A HREF="../../../../../org/encog/ml/genetic/genome/Chromosome.html" title="class in org.encog.ml.genetic.genome">Chromosome</A>&nbsp;links,
                  int&nbsp;inputCount,
                  int&nbsp;outputCount)</PRE>
<DL>
<DD>Create a NEAT gnome.
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>genomeID</CODE> - The genome id.<DD><CODE>neurons</CODE> - The neurons.<DD><CODE>links</CODE> - The links.<DD><CODE>inputCount</CODE> - The input count.<DD><CODE>outputCount</CODE> - The output count.</DL>
</DL>
<HR>

<A NAME="NEATGenome(long, int, int)"><!-- --></A><H3>
NEATGenome</H3>
<PRE>
public <B>NEATGenome</B>(long&nbsp;id,
                  int&nbsp;inputCount,
                  int&nbsp;outputCount)</PRE>
<DL>
<DD>Construct a genome, do not provide links and neurons.
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>id</CODE> - The genome id.<DD><CODE>inputCount</CODE> - The input count.<DD><CODE>outputCount</CODE> - The output count.</DL>
</DL>
<HR>

<A NAME="NEATGenome()"><!-- --></A><H3>
NEATGenome</H3>
<PRE>
public <B>NEATGenome</B>()</PRE>
<DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="alreadyHaveThisNeuronID(long)"><!-- --></A><H3>
alreadyHaveThisNeuronID</H3>
<PRE>
public boolean <B>alreadyHaveThisNeuronID</B>(long&nbsp;id)</PRE>
<DL>
<DD>Do we already have this neuron id?
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>id</CODE> - The id to check for.
<DT><B>Returns:</B><DD>True if we already have this neuron id.</DL>
</DD>
</DL>
<HR>

<A NAME="decode()"><!-- --></A><H3>
decode</H3>
<PRE>
public void <B>decode</B>()</PRE>
<DL>
<DD>Convert the genes to an actual network.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../org/encog/ml/genetic/genome/Genome.html#decode()">decode</A></CODE> in interface <CODE><A HREF="../../../../../org/encog/ml/genetic/genome/Genome.html" title="interface in org.encog.ml.genetic.genome">Genome</A></CODE></DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="encode()"><!-- --></A><H3>
encode</H3>
<PRE>
public void <B>encode</B>()</PRE>
<DL>
<DD>Convert the network to genes. Not currently supported.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../org/encog/ml/genetic/genome/Genome.html#encode()">encode</A></CODE> in interface <CODE><A HREF="../../../../../org/encog/ml/genetic/genome/Genome.html" title="interface in org.encog.ml.genetic.genome">Genome</A></CODE></DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getCompatibilityScore(org.encog.neural.neat.training.NEATGenome)"><!-- --></A><H3>
getCompatibilityScore</H3>
<PRE>
public double <B>getCompatibilityScore</B>(<A HREF="../../../../../org/encog/neural/neat/training/NEATGenome.html" title="class in org.encog.neural.neat.training">NEATGenome</A>&nbsp;genome)</PRE>
<DL>
<DD>Get the compatibility score with another genome. Used to determine
 species.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>genome</CODE> - The other genome.
<DT><B>Returns:</B><DD>The score.</DL>
</DD>
</DL>
<HR>

<A NAME="getInputCount()"><!-- --></A><H3>
getInputCount</H3>
<PRE>
public int <B>getInputCount</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>The number of input neurons.</DL>
</DD>
</DL>
<HR>

<A NAME="getLinks()"><!-- --></A><H3>
getLinks</H3>
<PRE>
public <A HREF="../../../../../org/encog/ml/genetic/genome/Chromosome.html" title="class in org.encog.ml.genetic.genome">Chromosome</A> <B>getLinks</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>THe links chromosome.</DL>
</DD>
</DL>
<HR>

<A NAME="getNetworkDepth()"><!-- --></A><H3>
getNetworkDepth</H3>
<PRE>
public int <B>getNetworkDepth</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>The network depth.</DL>
</DD>
</DL>
<HR>

<A NAME="getNeurons()"><!-- --></A><H3>
getNeurons</H3>
<PRE>
public <A HREF="../../../../../org/encog/ml/genetic/genome/Chromosome.html" title="class in org.encog.ml.genetic.genome">Chromosome</A> <B>getNeurons</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>The neurons chromosome.</DL>
</DD>
</DL>
<HR>

<A NAME="getNumGenes()"><!-- --></A><H3>
getNumGenes</H3>
<PRE>
public int <B>getNumGenes</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>The number of genes in the links chromosome.</DL>
</DD>
</DL>
<HR>

<A NAME="getOutputCount()"><!-- --></A><H3>
getOutputCount</H3>
<PRE>
public int <B>getOutputCount</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>The output count.</DL>
</DD>
</DL>
<HR>

<A NAME="getSpeciesID()"><!-- --></A><H3>
getSpeciesID</H3>
<PRE>
public long <B>getSpeciesID</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>The species ID.</DL>
</DD>
</DL>
<HR>

<A NAME="getSplitY(int)"><!-- --></A><H3>
getSplitY</H3>
<PRE>
public double <B>getSplitY</B>(int&nbsp;nd)</PRE>
<DL>
<DD>Get the specified split y.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>nd</CODE> - The neuron.
<DT><B>Returns:</B><DD>The split y.</DL>
</DD>
</DL>
<HR>

<A NAME="isDuplicateLink(long, long)"><!-- --></A><H3>
isDuplicateLink</H3>
<PRE>
public boolean <B>isDuplicateLink</B>(long&nbsp;fromNeuronID,
                               long&nbsp;toNeuronID)</PRE>
<DL>
<DD>Determine if this is a duplicate link.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>fromNeuronID</CODE> - The from neuron id.<DD><CODE>toNeuronID</CODE> - The to neuron id.
<DT><B>Returns:</B><DD>True if this is a duplicate link.</DL>
</DD>
</DL>
<HR>

<A NAME="mutateActivationResponse(double, double)"><!-- --></A><H3>
mutateActivationResponse</H3>
<PRE>
public void <B>mutateActivationResponse</B>(double&nbsp;mutateRate,
                                     double&nbsp;maxPertubation)</PRE>
<DL>
<DD>Mutate the activation response.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>mutateRate</CODE> - The mutation rate.<DD><CODE>maxPertubation</CODE> - The maximum to perturb it by.</DL>
</DD>
</DL>
<HR>

<A NAME="mutateWeights(double, double, double)"><!-- --></A><H3>
mutateWeights</H3>
<PRE>
public void <B>mutateWeights</B>(double&nbsp;mutateRate,
                          double&nbsp;probNewMutate,
                          double&nbsp;maxPertubation)</PRE>
<DL>
<DD>Mutate the weights.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>mutateRate</CODE> - The mutation rate.<DD><CODE>probNewMutate</CODE> - The probability of a whole new weight.<DD><CODE>maxPertubation</CODE> - The max perturbation.</DL>
</DD>
</DL>
<HR>

<A NAME="setNetworkDepth(int)"><!-- --></A><H3>
setNetworkDepth</H3>
<PRE>
public void <B>setNetworkDepth</B>(int&nbsp;networkDepth)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>networkDepth</CODE> - the networkDepth to set</DL>
</DD>
</DL>
<HR>

<A NAME="setSpeciesID(long)"><!-- --></A><H3>
setSpeciesID</H3>
<PRE>
public void <B>setSpeciesID</B>(long&nbsp;species)</PRE>
<DL>
<DD>Set the species id.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>species</CODE> - The species id.</DL>
</DD>
</DL>
<HR>

<A NAME="sortGenes()"><!-- --></A><H3>
sortGenes</H3>
<PRE>
public void <B>sortGenes</B>()</PRE>
<DL>
<DD>Sort the genes.
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getLinksChromosome()"><!-- --></A><H3>
getLinksChromosome</H3>
<PRE>
public <A HREF="../../../../../org/encog/ml/genetic/genome/Chromosome.html" title="class in org.encog.ml.genetic.genome">Chromosome</A> <B>getLinksChromosome</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the linksChromosome</DL>
</DD>
</DL>
<HR>

<A NAME="setLinksChromosome(org.encog.ml.genetic.genome.Chromosome)"><!-- --></A><H3>
setLinksChromosome</H3>
<PRE>
public void <B>setLinksChromosome</B>(<A HREF="../../../../../org/encog/ml/genetic/genome/Chromosome.html" title="class in org.encog.ml.genetic.genome">Chromosome</A>&nbsp;linksChromosome)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>linksChromosome</CODE> - the linksChromosome to set</DL>
</DD>
</DL>
<HR>

<A NAME="getNeuronsChromosome()"><!-- --></A><H3>
getNeuronsChromosome</H3>
<PRE>
public <A HREF="../../../../../org/encog/ml/genetic/genome/Chromosome.html" title="class in org.encog.ml.genetic.genome">Chromosome</A> <B>getNeuronsChromosome</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>the neuronsChromosome</DL>
</DD>
</DL>
<HR>

<A NAME="setNeuronsChromosome(org.encog.ml.genetic.genome.Chromosome)"><!-- --></A><H3>
setNeuronsChromosome</H3>
<PRE>
public void <B>setNeuronsChromosome</B>(<A HREF="../../../../../org/encog/ml/genetic/genome/Chromosome.html" title="class in org.encog.ml.genetic.genome">Chromosome</A>&nbsp;neuronsChromosome)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>neuronsChromosome</CODE> - the neuronsChromosome to set</DL>
</DD>
</DL>
<HR>

<A NAME="setInputCount(int)"><!-- --></A><H3>
setInputCount</H3>
<PRE>
public void <B>setInputCount</B>(int&nbsp;inputCount)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>inputCount</CODE> - the inputCount to set</DL>
</DD>
</DL>
<HR>

<A NAME="setOutputCount(int)"><!-- --></A><H3>
setOutputCount</H3>
<PRE>
public void <B>setOutputCount</B>(int&nbsp;outputCount)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>outputCount</CODE> - the outputCount to set</DL>
</DD>
</DL>
<HR>

<A NAME="validate()"><!-- --></A><H3>
validate</H3>
<PRE>
public void <B>validate</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/NEATGenome.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV CLASS&nbsp;
&nbsp;<A HREF="../../../../../org/encog/neural/neat/training/NEATInnovation.html" title="class in org.encog.neural.neat.training"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../index.html?org/encog/neural/neat/training/NEATGenome.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="NEATGenome.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#field_summary">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;<A HREF="#field_detail">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>
Copyright &#169; 2012. All Rights Reserved.
</BODY>
</HTML>
